VikingQuiz.UI/src/Components/ForgotPasswordComponent/ResetPasswordComponent/ResetPasswordComponent.tsx

```diff
@@ -0,0 +1,98 @@
+{
+    private readonly apiAddress: string = '/email';
+    private httpService: HttpService = new HttpService();
+    private storageService: StorageService = new StorageService();
+
+    constructor(props: any) {
+        super(props);
+
+        this.state = {
+            redirect: false,
+            showStatusMessage: false,
+            statusClass: '',
+            statusMessage: ''
+        };
+
+        const resetToken: string = this.props.match.params.token;
+        this.storageService.saveItem('token', resetToken);
+    }
+
+    public render() {
+        if (this.state.redirect) {
+            return (<Redirect to='/login' />);
+        }
+        return (
+            <div className='my-container'>
+                <HomeButton />
+                <div className='yellow-bg-message'>update password</div>
+                <div className='form-container'>
+                    <FormComponent className='signupForm' inputs={
+                        [
+                            new InputData('password', 'password', 'new password', '', 'password', ''),
+                            new InputData('confirm-password', 'password', 'confirm password', '', 'confirm-password', '')
+                        ]}
+                        url={this.apiAddress}
+                        buttonName=''
+                        onSubmit={this.onClickHandler}
+                        validator={changePasswordValidator}
+                        validationRules={changeRule}
+                    />
+                    {this.state.showStatusMessage ? (<div className={`message ${this.state.statusClass}`}>{this.state.statusMessage}</div>) : null}
+                </div>
+            </div>
+        );
+    }
+
+    private onClickHandler = (url: string, formData: any) => {
+        if (!url || !formData) {
+            return;
+        }
+
+        const axiosBody: any = { 'password': formData.password };
+        this.setState({ showStatusMessage: true });
+        this.httpService.putWithToken(url, axiosBody)
+            .then(() => this.resetPasswordSuccess())
+            .catch((error: any) => this.resetPasswordError(error));
+    }
+
+    private resetPasswordSuccess = () => {
+        this.setState({
+            statusClass: 'success-message',
+            statusMessage: 'Password changed, redirecting to Log In'
+        });
+        setTimeout(
+            () => this.setState({ redirect: true }),
+            3500
+        );
+    };
+
+    private resetPasswordError = (error: any) => {
+        this.setState({ statusClass: 'error-message' });
+        if (error.response === undefined) {
+            this.setState({ statusMessage: 'Could not connect to server. Please try again later' });
+        }
+        else {
+            // token is invalid (either expired or fake)
+            if (error.response.status === 401) {
+                this.setState({ statusMessage: 'This reset link is invalid has expired' });
```

Comment:
Proper text: 'This reset link is invalid has expired' should be "'This reset link is invalid or has expired.'"